sqlglot/tokens.py

```diff
@@ -1300,8 +1301,12 @@ def _scan_number(self) -> None:
                     self._add(TokenType.NUMBER, number_text)
                     self._add(TokenType.DCOLON, "::")
                     return self._add(token_type, literal)
-                elif self.dialect.IDENTIFIERS_CAN_START_WITH_DIGIT:
-                    return self._add(TokenType.VAR)
+                else:
+                    replaced = literal.replace("_", "")
+                    if self.dialect.NUMBERS_CAN_BE_UNDERSCORE_SEPARATED and replaced.isdigit():
+                        return self._add(TokenType.NUMBER, number_text + replaced)
+                    elif self.dialect.IDENTIFIERS_CAN_START_WITH_DIGIT:
```

Comment:
```suggestion
                    if self.dialect.IDENTIFIERS_CAN_START_WITH_DIGIT:
```

Reply:
Hm these should be mut-ex, no (?)

Reply:
If you enter the first branch you return, they're still mutually-exclusive. This style is just more consistent with what we do in the codebase in general.

Reply:
Right, deja vu